### DISCUSSION TITLE
Deploying Laravel application using `serversideup/docker-php` is so easy 🙌

### DISCUSSION THREAD

**Discussion by aagjalpankaj** [serversideup/docker-php#466] [Show and tell]
We had a Laravel application hosted on-premises and we wanted to migrate to AWS App-runner.

### What we did:
- Added `Dockerfile` under directory `docker/application` as below
```
### BASE: things required for all - dev, ci, prod.
FROM serversideup/php:8.1-fpm-nginx-alpine-v3.0.0 AS base

USER root
RUN install-php-extensions intl bcmath

### BUILD: things required for dev & ci.
FROM base AS build

RUN apk add --no-cache --virtual .build-deps g++ make \
    && apk add --no-cache nodejs npm python3 \
    && ln -sf /usr/bin/python3 /usr/bin/python \
    && apk del .build-deps

### CI: specific to ci (if any)
FROM build AS ci

RUN apk add --no-cache mysql-client

USER www-data

### DEV: specific to local (if any)
FROM build AS dev

USER www-data

### PROD - for deployment
FROM base AS prod

ENV AUTORUN_ENABLED="true" \
    AUTORUN_LARAVEL_MIGRATION_ISOLATION="false" \
    AUTORUN_LARAVEL_CONFIG_CACHE="false" \
    PHP_OPCACHE_ENABLE="1"

USER www-data

```
- Pushed ci, dev & prod images with version v.0.0.1 to self-hosted GitLab registry as per above `Dockerfile`.
- Added `docker-compose.yml` which uses `dev:v0.0.1`
- Added `gitlab-ci.yml` with CI/CD job which uses `ci:v0.0.1`
- Added another `Dockerfile` at root-level as below which uses `prod:v0.0.1` for actual deployment from `.gitlab-ci.yml` deploy job
```
FROM <gitlab-registry-url>/prod:v0.0.1

COPY --chown=www-data:www-data . /var/www/html

STOPSIGNAL SIGTERM
CMD ["php", "/var/www/html/artisan", "schedule:work"]

USER www-data

```

Below is the image size:
- ci: 102MB
- dev: 84MB
- prod: 53MB

---

`serversideup/docker-php` made it easy for us to deploy a Laravel application right out of the box, without having to worry about low-level configuration. Thank you so much! 🙂

What do you think of this approach? Any suggestions or improvements would really help.

**Comment by jaydrogers** [comment]
Thank you so much for your compliments! I am very happy things are working out for you 😃
